src/robotFactory.js

```diff
@@ -1,10 +1,87 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = coords;
```

Comment:
You can simplify it like this and delete lines 10-17.
```suggestion
    this.coords = {
      x: coords.x || 0,
      y: coords.y || 0,
    };
```